File: time_needed_to_buy_tickets.go
func timeRequiredToBuy(tickets []int, k int) int {
    var ans int = 0;
        
    for index, val := range tickets {
        if index > k {
            ans += min(tickets[k]-1, val);
        } else {
            ans += min(tickets[k], val);
        }
    }
    return ans; 
}

// Time Complexity : O(n)
// Space Complexity : O(1)

File: time_needed_to_buy_tickets.cpp
class Solution {
public:
    int timeRequiredToBuy(vector<int>& tickets, int k) {
        int ans = 0;
        
        for(int i = 0; i < tickets.size(); i++) {
            if(i > k) {
                ans += min(tickets[k]-1, tickets[i]);
            } else {
                ans += min(tickets[k], tickets[i]);
            }
        }
        return ans;           
    }
};

// Time Complexity : O(n)
// Space Complexity : O(1)
